lib/router.js

```diff
@@ -98,8 +101,10 @@ Router.prototype._handleRESTBasePathSpec = function(node, subspec, symbols, uri)
             // set up a handler
             // XXX: share?
             node.value.methods[methodName] = function (restbase, req) {
-                // XXX: expand the request with req information!
-                restbase.request(mxrb.service);
+                function eval_req(req) {
```

Comment:
It would be nice if you could move request_parser out, and return an eval closure that takes the request from there. In the handler, you can then just call it.